### PR TITLE
[bitcoin-core] Support AFL builds

### PR DESCRIPTION
**Pull request by guidovranken**
CC @MarcoFalke

### CODE REVIEW

**Review by DavidKorczynski — APPROVED** [review]
This looks good to me, thanks Guido! Since this is a fairly new project and an issue related to it was opened here https://github.com/google/oss-fuzz/issues/5713 I will just wait for @MarcoFalke approval before merging.

File: projects/bitcoin-core/build.sh
Comment (MarcoFalke) [inline projects/bitcoin-core/build.sh:45]: I was hoping to get rid of this special case and replace this with 
```suggestion
  CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable-fuzz SANITIZER_LDFLAGS="$LIB_FUZZING_ENGINE"
```

as well

**Review by MarcoFalke — APPROVED** [review]
Thanks. Looks good. Some minor cleanup ideas that can be done the next time the file is touched.

**Comment by MarcoFalke** [comment]
Suggested cleanup diff:

```diff
diff --git a/projects/bitcoin-core/build.sh b/projects/bitcoin-core/build.sh
index 92cd4f94..30fb71d1 100755
--- a/projects/bitcoin-core/build.sh
+++ b/projects/bitcoin-core/build.sh
@@ -36,17 +36,13 @@ fi
 
 # Build the fuzz targets
 
+sed -i "s|PROVIDE_FUZZ_MAIN_FUNCTION|NEVER_PROVIDE_MAIN_FOR_OSS_FUZZ|g" "./configure.ac"
 ./autogen.sh
 
 # OSS-Fuzz will provide CC, CXX, etc. So only set:
 # * --enable-fuzz, see https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md
 # * CONFIG_SITE, see https://github.com/bitcoin/bitcoin/blob/master/depends/README.md
-if [ "$FUZZING_ENGINE" = "libfuzzer" ]; then
-  CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable-fuzz --with-sanitizers=fuzzer
-else
-  sed -i "s|PROVIDE_FUZZ_MAIN_FUNCTION|NEVER_PROVIDE_MAIN_FOR_OSS_FUZZ|g" "./src/test/fuzz/fuzz.cpp"
-  CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable-fuzz SANITIZER_LDFLAGS="$LIB_FUZZING_ENGINE"
-fi
+CONFIG_SITE="$PWD/depends/$BUILD_TRIPLET/share/config.site" ./configure --enable-fuzz SANITIZER_LDFLAGS="$LIB_FUZZING_ENGINE"
 
 make -j$(nproc)
 
@@ -55,7 +51,6 @@ FUZZ_TARGETS=( 'process_messages' 'asmap' )
 for fuzz_target in ${FUZZ_TARGETS[@]}; do
   git checkout -- "./src/test/fuzz/fuzz.cpp"
   sed -i "s|std::getenv(\"FUZZ\")|\"$fuzz_target\"|g" "./src/test/fuzz/fuzz.cpp"
-  sed -i "s|PROVIDE_FUZZ_MAIN_FUNCTION|NEVER_PROVIDE_MAIN_FOR_OSS_FUZZ|g" "./src/test/fuzz/fuzz.cpp"
   make -j$(nproc)
   mv ./src/test/fuzz/fuzz $OUT/$fuzz_target
   (

**Comment by MarcoFalke** [comment]
@jonathanmetzman You can merge this one

**Pull request merged**
